internal/expr/expr.go

```diff
@@ -0,0 +1,106 @@
+package expr
+
+import (
+	"errors"
+	"go/constant"
+	"go/token"
+	"go/types"
+)
+
+var ErrEvalIsUnknown = errors.New("expression is unknown")
+var ErrEvalNotBool = errors.New("expression does not return a bool expression")
+var ErrEvalNotString = errors.New("expression does not return a string expression")
+
+// parse the constant string via a token evaluation
+func eval(s string) (constant.Value, error) {
+	tv, err := types.Eval(token.NewFileSet(), nil, token.NoPos, s)
+	if err != nil {
+		return nil, err
+	}
+	if tv.Value.Kind() == constant.Unknown {
+		return nil, ErrEvalIsUnknown
+	}
+	return tv.Value, nil
+}
+
+// ToFloat32 evaluates the given string as a float32, or it returns an error if the result is invalid or the expression is Unknown.
+func ToFloat32(s string) (float32, error) {
+	val, err := eval(s)
+	if err != nil {
+		return 0, err
+	}
+	res, _ := constant.Float32Val(val)
+	return res, nil
+}
+
+// ToFloat64 evaluates the given string as a float64, or it returns an error if the result is invalid or the expression is Unknown.
+func ToFloat64(s string) (float64, error) {
+	val, err := eval(s)
+	if err != nil {
+		return 0, err
+	}
+	res, _ := constant.Float64Val(val)
```

Comment:
Does it make sense to validate the type before getting the value, to avoid possible panic?

Reply:
I think, to some extent this is already handled by the underlying code.

e.g.
https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/go/constant/value.go;l=548

Reply:
but yes, there's a panic in the.